src/controllers/dynakube/dynakube_phase.go

```diff
@@ -0,0 +1,86 @@
+package dynakube
+
+import (
+	"context"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/activegate/capability"
+	appsv1 "k8s.io/api/apps/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (controller *DynakubeController) determineDynaKubePhase(instance *dynatracev1beta1.DynaKube) bool {
+	if instance.NeedsActiveGate() {
+		activeGatePods, err := controller.numberOfMissingActiveGatePods(instance)
+		if err != nil {
+			log.Error(err, "activegate sts could not be accessed", "dynakube", instance.Name)
+			return updatePhaseIfChanged(instance, dynatracev1beta1.Error)
```

Comment:
wouldn't something similar to `status.SetPhaseOnError` work better here?
Could hide implementation details for the other 3 phases and have the phase logic in a single file instead of split.